test_can_bundle_configured_catalogs fails with specific seed

With this patch …

```diff
Subject: [PATCH] WIP
---
Index: test/integration_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/integration_test.py b/test/integration_test.py
--- a/test/integration_test.py	(revision 3f8d8d9fb3441bbbfb05fa766ec8d97e82b8b59c)
+++ b/test/integration_test.py	(date 1740503707311)
@@ -236,7 +236,7 @@
         super().setUp()
         # All random operations should be made using this seed so that test
         # results are deterministically reproducible
-        self.random_seed = randint(0, sys.maxsize)
+        self.random_seed = 6071700547648562554 # randint(0, sys.maxsize)
         self.random = Random(self.random_seed)
         log.info('Using random seed %r', self.random_seed)
 

```

… the test fails with …

```
SubTest failure: Traceback (most recent call last):
  File "/Users/hannes/.pyenv/versions/3.12.8/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/Users/hannes/.pyenv/versions/3.12.8/lib/python3.12/unittest/case.py", line 539, in subTest
    yield
  File "/Users/hannes/workspace/hca/azul.hannes2.local/test/integration_test.py", line 1841, in test_can_bundle_configured_catalogs
    self._test_catalog(catalog)
  File "/Users/hannes/workspace/hca/azul.hannes2.local/test/integration_test.py", line 1832, in _test_catalog
    self.assertEqual([], links)
  File "/Users/hannes/Applications/PyCharm Professional Edition.app/Contents/plugins/python-ce/helpers/pycharm/teamcity/diff_tools.py", line 33, in _patched_equals
    old(self, first, second, msg)
AssertionError: Lists differ: [] != [{'inputs': ['anvil_dataset/677dd55c-3fa3-[54 chars] []}]

Second list contains 1 additional elements.
First extra element 0:
{'inputs': ['anvil_dataset/677dd55c-3fa3-4b07-8c98-985d94d7577e'], 'activity': None, 'outputs': []}

- []
+ [{'activity': None,
+   'inputs': ['anvil_dataset/677dd55c-3fa3-4b07-8c98-985d94d7577e'],
+   'outputs': []}]

```

… in my `anvilbox`-based personal deployment. 

The test was run with `develop` checked out (3f8d8d9f) but that commit wasn't deployed, some other branch (ahead of develop) was. I don't think the deployment is involved in this test so it shouldn't matter, but assignee should first attempt to reproduce.

The purpose of the conditional `if len(entities) > 1` isn't immediately obvious to me and should be explained in a comment. I also suspect that the case `len(entities) == 1` and potentially even `len(entities) == 0` need more assertions. For example, what is that singleton entity? In fact , those assertions might even eliminate the need for an explanatory comment.

Reproduced on the latest develop (0b53a96cc). I did not check out the commit references in the issue because doing so would have required reverting to an older version of python (recreating the virtual environment, etc).

```diff
Subject: [PATCH] PATCH
---
Index: test/integration_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/integration_test.py b/test/integration_test.py
--- a/test/integration_test.py	(revision 0b53a96cca4539d22255f894605004f83f0b8574)
+++ b/test/integration_test.py	(date 1741042307118)
@@ -236,7 +236,7 @@
         super().setUp()
         # All random operations should be made using this seed so that test
         # results are deterministically reproducible
-        self.random_seed = randint(0, sys.maxsize)
+        self.random_seed = 6071700547648562554 # randint(0, sys.maxsize)
         self.random = Random(self.random_seed)
         log.info('Using random seed %r', self.random_seed)
```

```
AssertionError: Lists differ: [] != [{'inputs': ['anvil_dataset/677dd55c-3fa3-[54 chars] []}]

Second list contains 1 additional elements.
First extra element 0:
{'inputs': ['anvil_dataset/677dd55c-3fa3-4b07-8c98-985d94d7577e'], 'activity': None, 'outputs': []}

- []
+ [{'activity': None,
+   'inputs': ['anvil_dataset/677dd55c-3fa3-4b07-8c98-985d94d7577e'],
+   'outputs': []}]



One or more subtests failed
Failed subtests list: (catalog='anvil-it', repository=Config_Catalog_Plugin(name='tdr_anvil'))
```

--- a/src/azul/plugins/repository/tdr_anvil/__init__.py
+++ b/src/azul/plugins/repository/tdr_anvil/__init__.py
@@ -472,10 +472,15 @@ class Plugin(TDRPlugin[TDRAnvilBundle, TDRAnvilBundleFQID]):
                 linked_file_refs.add(file_ref)
         dataset_ref, dataset_row = self._get_dataset(source)
         result.add_entity(dataset_ref, self._version, dict(dataset_row))
-        result.add_links([
-            EntityLink(inputs=singleton(dataset_ref),
-                       outputs=frozenset(linked_file_refs))
-        ])
+        # Avoid inserting "degenerate" links with an empty list of outputs, i.e.
+        # in case of an empty batch (as is common on `anvilbox`). Such links
+        # would be harmless in production, but would complicate the bundle
+        # canning integration test.
+        if linked_file_refs:
+            result.add_links([
+                EntityLink(inputs=singleton(dataset_ref),
+                           outputs=frozenset(linked_file_refs))
+            ])
         return result
 
     def _duos_bundle(self, bundle_fqid: TDRAnvilBundleFQID) -> TDRAnvilBundle:

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -1861,14 +1861,13 @@ class CanBundleScriptIntegrationTest(IntegrationTestCase):
                 self.assertIsInstance(entities, dict)
                 self.assertIsInstance(links, list)
                 entities = set(map(EntityReference.parse, entities.keys()))
-                if len(entities) > 1:
-                    linked_entities = frozenset().union(*(
-                        EntityLink.from_json(link).all_entities
-                        for link in links
-                    ))
-                    self.assertEqual(entities, linked_entities)
-                else:
-                    self.assertEqual([], links)
+                for link in map(EntityLink.from_json, links):
+                    self.assertGreater(len(link.inputs), 0)
+                    self.assertGreater(len(link.outputs), 0)
+                    # Since we know the links' inputs and outputs are nonempty,
+                    # this also validates that bundles containing only orphans
+                    # contain no links.
+                    self.assertIsSubset(link.all_entities, entities)
             else:
                 assert False, metadata_plugin_name
 

